test/test_utils.py

```diff
@@ -112,6 +114,76 @@ def __len__(self):
         return 10
 
 
+class TestCheckpoint(TestCase):
+
+    def test_checkpoint_valid(self):
+        model = nn.Sequential(
+            nn.Linear(100, 50),
+            nn.ReLU(),
+            nn.Linear(50, 20),
+            nn.ReLU(),
+            nn.Linear(20, 5),
+            nn.ReLU()
+        )
+
+        input_var = Variable(torch.randn(1, 100), requires_grad=True)
+
+        # checkpointed
+        chunks = 2
+        modules = [module for k, module in model._modules.items()]
+        out = checkpoint_sequential(modules, chunks, input_var)
+        with self.assertRaises(AssertionError):
+            torch.autograd.grad(
+                outputs=[out], grad_outputs=[torch.ones(1, 5)], inputs=[input_var], create_graph=True
+            )
+
+    def test_checkpoint(self):
```

Comment:
This test looks like a correctness check, but it does not actually check that checkpointing has happened. Is it possible to check for this too? Related: https://github.com/pytorch/pytorch/issues/6010

Reply:
yep, this was correctness check that we the checkpointed and non-checkpointed models have same output/grads. 
for checking whether checkpointing has happened: how about passing a bool arg to the checkpoint call and assert there that the bool is received and true?

Reply:
done